### PR TITLE
Feat : improving etmp and idms file reader  methods to ensure file is…

### PR DESCRIPTION
**Pull request by thadeusssewanyana**
… closed after reading contents

### CODE REVIEW

**Review by sainahjan — CHANGES_REQUESTED** [review]
Left a couple of comments. I don't think the implementation is tidy/correct. There might be better ways of unwrapping a Try with custom errors than `fold`, but `getOrElse` is definitely not the right way to do it because it discards the Failure.

If you don't want to modify the error and you really want to throw, `.get` on a Try has the desired behaviour.

Also, avoid using the operator syntax with regular methods. The Scala designers have even deprecated the feature in Scala 3 because they agreed that it makes the code uneradable.

Don't use the operator syntax on braces, if you can avoid it.
e.g.
```scala
// Avoid:
something { bla bla
  bla
} anotherCall { bla bla bla bla bla ... } andAnother { bla =>
  bla
  bla
}
// Prefer:
val somethingNew = something { bla bla
  bla
  }

somethingNew
  .anotherCall { bla bla bla bla bla ... }
  .andAnother { bla =>
    bla
    bla
  }

// Or:
something { bla bla
    bla
  }
  .anotherCall { bla bla bla bla bla ... }
  .andAnother { bla =>
    bla
    bla
  }
```

Basically, avoid putting the top-level syntactic elements in the middle of larger secondary syntactic elements.

**Review by sainahjan — APPROVED** [review]

**Review by rogerball — APPROVED** [review]
